airflow/models/taskinstance.py

```diff
@@ -3676,12 +3676,14 @@ def _schedule_downstream_tasks(
             # Get a partial DAG with just the specific tasks we want to examine.
             # In order for dep checks to work correctly, we include ourself (so
             # TriggerRuleDep can check the state of the task we just executed).
-            partial_dag = task.dag.partial_subset(
-                task.downstream_task_ids,
-                include_downstream=True,
-                include_upstream=False,
-                include_direct_upstream=True,
-            )
+            # partial_dag = task.dag.partial_subset(
+            #     task.downstream_task_ids,
+            #     include_downstream=True,
+            #     include_upstream=False,
+            #     include_direct_upstream=True,
+            # )
+            # Getting partial_dag is an unsafe behavior
+            partial_dag = task.dag
 
             dag_run.dag = partial_dag
```

Comment:
If we remove it completely, it would be better to remove the `partial_dag` variable and fix commentary accordingly :)

Reply:
I have completely removed `partial_dag` and then fixed the tests